nsdb-core/src/main/scala/io/radicalbit/nsdb/actors/MetricReaderActor.scala

```diff
@@ -161,7 +161,9 @@ class MetricReaderActor(val basePath: String, nodeName: String, val db: String,
       postProcFun: Seq[Bit] => Seq[Bit]): Future[Either[SelectStatementFailed, Seq[Bit]]] = {
     Future
       .sequence(actors.map {
-        case (_, actor) => actor ? ExecuteSelectStatement(statement, schema, actors.map(_._1))
+        case (_, actor) =>
+          (actor ? ExecuteSelectStatement(statement, schema, actors.map(_._1)))
+            .recoverWith { case t => Future(SelectStatementFailed(t.getMessage)) }
```

Comment:
What about using recover and avoid to wrap the `SelectStatementField` with a `Future`?